sources/packages/web/src/services/AuthService.ts

```diff
@@ -144,51 +113,89 @@ export class AuthService {
-      return false;
-    } else if (await UserService.shared.checkUser(authHeader)) {
-      if (!(await UserService.shared.checkActiveUser(authHeader))) {
-        await this.logout(ClientIdType.Institution, { isUserDisabled: true });
-        return true;
-      }
-      await store.dispatch("institution/initialize", authHeader);
-      return false;
+  /**
+   * Process the login for the student redirecting to the student profile
+   * creation case it is the first access.
+   */
+  private async processStudentLogin() {
+    const studentStore = useStudentStore(store);
+    const hasStudentAccount =
+      await StudentService.shared.synchronizeFromUserToken();
+    await studentStore.setHasStudentAccount(hasStudentAccount);
+    if (hasStudentAccount) {
+      await studentStore.updateProfileData();
     } else {
-      if (
-        await InstitutionService.shared.checkIfExist(
-          bceIdAccountDetails.institution.guid,
-          authHeader,
-        )
-      ) {
-        await this.logout(ClientIdType.Institution, { isUnknownUser: true });
-        return true;
-      }
+      // If the student is not present, redirect to student profile
+      // for account creation.
+      this.priorityRedirect = {
+        name: StudentRoutesConst.STUDENT_PROFILE,
+      };
+    }
+  }
+
+  /**
+   * Process the login for the institution executing the verifications
+   * to determine if the user can proceed or must be redirect somewhere.
+   */
+  private async processInstitutionLogin() {
+    const userStatus =
+      await InstitutionService.shared.getInstitutionUserStatus();
+    if (userStatus.isActiveUser === true) {
+      // User is active so just proceed.
+      await store.dispatch("institution/initialize");
+      return;
+    }
+
+    if (userStatus.isActiveUser === false) {
+      // User exists and it is not active.
+      // Redirect to login page with a proper message.
+      await this.logout(ClientIdType.Institution, {
+        isUserDisabled: true,
+      });
+      return;
+    }
+
+    if (
+      !userStatus.isExistingUser &&
+      !userStatus.associatedInstitutionExists &&
+      userStatus.hasBusinessBCeIDAccount
+    ) {
+      // Business BCeID user logging for the first time when the institution is not created yet.
+      // Redirect to the institution profile to allow the user to create the institution.
       this.priorityRedirect = {
         name: InstitutionRoutesConst.INSTITUTION_CREATE,
       };
-      return false;
+      return;
+    }
+
+    if (!userStatus.isExistingUser) {
+      // User is a business BCeID, the institution exists but the user was never added, hence he is not allowed to access
+      // the application and must request to his administrator to include him using the manage users UI.
+      // Or user is a basic BCeID that must be added by the Ministry before have access to the system.
+      await this.logout(ClientIdType.Institution, {
+        isUnknownUser: true,
+      });
```

Comment:
Are we missing a return here?. Also IMO, as all the conditions revolve around the userStatus,  instead of having if, it should be better to go with if and else and a single return.

Reply:
I was not able to follow where the return would be missing, do you mind elaborating, please?
The way that the code is organized is supposed to be more clear and also follow [this pattern](https://medium.com/swlh/return-early-pattern-3d18a41bba8) which is something that we talked about in the past.
Doing if/else and nesting them is exactly what we try to avoid and also what Sonarcloud with be pointing auto.